### PR TITLE
Change categories in Module Manager

### LINKED ISSUE #10048: [FF-177] [Module Manager] Categories
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/FF-177).

- _**Reporter:**_ lou
- _**Created at:**_ Mon, 28 May 2018 11:56:14 +0200

<p><b>USER STORY</b></p>

<p>As a merchant,  I want to easily manage and configure my modules.</p>

<p><b>Customer/business value (why)</b></p>

<p>Module management is a key part for merchants and us to make revenue. This page has been changed a lot of time and we still need to improve it. in this page we talk about Native modules, but they are not well undestood by merchants. We aim to removing this word from the page and we use the same categories as used on Addons marketplace.</p>

<p><b>WHAT</b></p>

<p>Module Manager corresponds to the current page "Installed Modules". The changes are described below :</p>
<ul>
	<li><ins>New Hierarchy</ins>: Delete the categories "native modules" and "installed modules" and replace with :
	<ul>
		<li>the 10 categories (Design & Navigation, Product Page... named "Addons categories N1")</li>
		<li>+ Category "Other" : includes all customized modules and modules that are not sorted out in one of the 11 categories (10 Addons categories N1 + 1 theme modules)</li>
		<li>+ "Recently used" : includes the 10 <b>lastest used</b> modules</li>
		<li>Keep the category "Theme modules"</li>
	</ul>
	</li>
</ul>

<p> </p>
<ul>
	<li><ins>Category order on the page:</ins> There are 12 categories on the page. By default, the categories order <b>ON THE PAGE</b> will always be (don't take the order of the mock up)
	<ul>
		<li>Administration</li>
		<li>Design & Navigation</li>
		<li>Promotions & Marketing</li>
		<li>Product Page</li>
		<li>Traffic & Marketplaces</li>
		<li>Shipping & Logistics</li>
		<li>Payment</li>
		<li>Customers</li>
		<li>Facebook & Social Networks</li>
		<li>Specialized Platforms</li>
		<li>Theme modules</li>
		<li>Other</li>
	</ul>
	</li>
</ul>

<p>Note that "recently used does not appear, unless it has been selected in the "select" filter.</p>

<p> </p>
<ul>
	<li><ins>Category sorting:</ins> Modules will be sorted out by Addons API. If there is no connexion to Addons, only 2 categories will be displayed (My modules and Theme modules). My modules will include all modules except those with the feature "theme module". This category appears only if API is down.</li>
</ul>

<p> </p>
<ul>
	<li><ins>Category selection</ins>: In the select "Categories", there are 13 categories.
	<ul>
		<li>The order is : "Recently used" then the 12 categories in order as displayed before. </li>
		<li>The filter is mono-select.</li>
		<li>When "Recently used" is selected, it will appear in the same way as we selected one of the 12 categories. </li>
	</ul>
	</li>
</ul>

<p> </p>
<ul>
	<li><ins>Last Access:</ins>
	<ul>
		<li>Inside each category, the modules must be sorted by "last access".</li>
		<li>Remove the sorting "last access" and "A-Z"</li>
		<li>When a module has been installed (paid or free), it must be considered as latest accessed then it should appear at the top of the category, even if the merchant did not configure it (only installed it)</li>
	</ul>
	</li>
</ul>

<p> </p>
<ul>
	<li><ins>0 product in this category:</ins></li>
</ul>

<p><b>Only for 10 Addons category N1</b> : If there are not any products in the category, then show a window saying "<em>There are no modules "<span class="error">&#91;NAME OF THE CATEGORY N1&#93;</span>". Discover the best-selling modules of this category in the "Add New module" page</em>". There is a link on "Add new module" that returns to the Add new modules tab with the category N1 pre-selected. Cf MOCK UP.</p>
<ul>
	<li><ins>See more:</ins></li>
</ul>

<p>Inside each of the 12 categories (no Recently used since it's limited to 10 modules), add a "see more" when the category has more than 6 modules (>6 modules) and a "see less" to come back to the exclusive list of modules.</p>
<ul>
	<li><ins>Search:</ins></li>
</ul>

<p>When the used searches for something in the search bar, show the modules that fit the search and show the categories. </p>
<ul>
	<li> <ins>General design</ins>
	<ul>
		<li>font, titles ... see the mock up</li>
		<li>location of the select windows and search bar (see the mock up)</li>
	</ul>
	</li>
</ul>

<p> </p>

<p>MOCK UP <a href="https://projects.invisionapp.com/share/AJMX15FFGD4#/screens/308311801" class="external-link" rel="nofollow">https://projects.invisionapp.com/share/AJMX15FFGD4#/screens/308311801</a></p>
        
* How to reproduce the issue ?

N/A

### LINKED ISSUE #10251: Notifications icons not correctly placed on module page
> **Reminder**
> 
> DO NOT disclose security issues here, contact security@prestashop.com instead!

**Describe the bug**
Notifications icons are not correctly placed on module page

**To Reproduce**
1. Go to Modules > Modules manager
2. Look notifications icons on alerts & updates tabs

**Screenshots**
capture d ecran_226

Should be placed like this:

capture d ecran_227

**Additionnal information**
PrestaShop version: develop
PHP version: N/A

### LINKED ISSUE #10460: When scrolling buttons appears on the top layer
**Describe the bug**
When you scroll on Modules pages (module catalog or module manager) you can see "search" button appearing under the top layer header toolbar.

**To Reproduce**
1. Go to 'Modules > module catalog'
2. Scroll down
3. Look at the search button => it appears under the header toolbar

**Screenshots**
https://drive.google.com/file/d/1OOSeqmjbVdwWHRgSf_2aHxDct7_3ZXgH/view?usp=sharing

**Additionnal information**
PrestaShop version: Develop 1.7.5.0
PHP version: N/A

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Follow issue.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10048 <br/>Fixes #10251<br/>Fixes #10460
| How to test?  | Follow tickets.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10440)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php [hunk 1]
```diff
@@ -113,32 +113,26 @@ public function manageAction()
...
             [
+                'maxModulesDisplayed' => self::MAX_MODULES_DISPLAYED,
+                'bulkActions' => $bulkActions,
                 'layoutHeaderToolbarBtn' => $this->getToolbarButtons(),
-                'layoutTitle' => $this->trans('Manage installed modules', 'Admin.Modules.Feature'),
-                'modules' => $modules,
-                'topMenuData' => $this->getTopMenuData(
-                    $this->get('prestashop.categories_provider')->getCategoriesMenu($installedProducts)
-                ),
+                'layoutTitle' => $this->trans('Module manager', 'Admin.Modules.Feature'),
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php:133]: Wouldn't it be better to write _Modules manager_ in the plural, I assume that there are several modules managed here, right?

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_bulk.html.twig [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  *
+  * @author    PrestaShop SA <contact@prestashop.com>
+  * @copyright 2007-2018 PrestaShop SA
+  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+  * International Registered Trademark & Property of PrestaShop SA
+  *#}
+<div class="ps-dropdown dropdown btn-group bordered mb-1 module-bulk-actions disabled">
+  <div id="bulk-actions-dropdown" class="dropdown-label" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+    <span class="js-selected-item selected-item module-bulk-actions-selector-label">
+      {{ 'Uninstall'|trans({}, 'Admin.Modules.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_bulk.html.twig]: _Admin.Actions_ instead?

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig [hunk 3]
```diff
@@ -1,55 +1,65 @@
...
+    <div class="ps-dropdown-menu dropdown-menu module-category-selector" aria-labelledby="{{refMenu}}">
+      <ul class="items-list js-items-list">
+        <li class="module-category-reset">
+          <a class="dropdown-item" href="#">
+            {{ 'All Categories'|trans({}, 'Admin.Modules.Feature') }}
+          </a>
+        </li>
+        <li class="module-category-menu module-category-recently-used"
+          data-category-ref="recently-used"
+          data-category-display-name="{{ 'Recently Used'|trans({}, 'Admin.Modules.Feature') }}"
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig]: What is it supposed to mean?

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/menu_top.html.twig [hunk 4]
```diff
@@ -1,54 +1,69 @@
...
+      </div>
+    </div>
+    <div class="col-md-4 module-menu-item">
     </div>
+  </div>
+
+  <div class="row">
+    {% if topMenuData is defined %}
+      <div class="col-md-4 module-top-menu-item">
+        <h3>{{ 'Category' | trans({}, 'Admin.Modules.Feature') }}</h3>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/menu_top.html.twig]: I think it should be localized in _Admin.Catalog.Feature_.

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/menu_top.html.twig [hunk 5]
```diff
@@ -1,54 +1,69 @@
...
+        {% include 'PrestaShopBundle:Admin/Module/Includes:dropdown_categories.html.twig' with { 'topMenuData': topMenuData } %}
+      </div>
+    {% endif %}
+
+    {% if requireFilterStatus is defined and requireFilterStatus == true %}
+      <div class="col-md-4 module-top-menu-item">
+        <h3>{{ 'Status' | trans({}, 'Admin.Modules.Feature') }}</h3>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/menu_top.html.twig]: _Admin.Global_ here.

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  *
+  * @author    PrestaShop SA <contact@prestashop.com>
+  * @copyright 2007-2018 PrestaShop SA
+  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+  * International Registered Trademark & Property of PrestaShop SA
+  *#}
+<div class="col-12">
+  <p class="text-center">
+    <button type="button" class="btn btn-link see-more" data-category="{{ id }}">
+      {{ 'See more' | trans({}, 'Admin.Modules.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig]: It already exists in _Admin.Orderscustomers.Feature_ but I guess it should be smarter to make it fall into the _Admin.Actions_ file. Indeed, it is a generic action that can be found elsewhere in the back office.

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    </button>
+    <button type="button" class="btn btn-link see-less d-none" data-category="{{ id }}">
+      {{ 'See less' | trans({}, 'Admin.Modules.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig]: Same than the 'See more' string. :-)

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig [hunk 8]
```diff
@@ -1,107 +1,77 @@
...
-            {% include 'PrestaShopBundle:Admin/Module/Includes:grid_manage_installed.html.twig' with { 'modules': modules.native_modules, 'display_type': 'list', 'origin': 'manage', 'id': 'native' } %}
+      <div id="module-recently-used-list" class="module-short-list">
+        <span id="recently-used_modules" class="module-search-result-wording">{{ 'Recently Used' | trans({}, 'Admin.Modules.Feature') }}</span>
+        <div class="modules-list" data-name="recently-used"></div>
+      </div>
 
-            <hr class="top-menu-separator">
+      {% for category in categories.subMenu %}
+        <div class="module-short-list">
+          <span id="{{ category.tab }}_modules" class="module-search-result-wording">{{ category.name | trans({}, 'Admin.Modules.Feature') }}</span>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: What is the wording here, category.name?

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig [hunk 9]
```diff
@@ -1,107 +1,77 @@
...
-            <div class="module-short-list">
-                <span id="theme_modules" class="module-search-result-wording">{{ '%nbModules% theme modules'|trans({'%nbModules%' : modules.theme_bundle|length}, 'Admin.Modules.Feature') }}</span>
-                <span class="help-box" data-toggle="popover"
-                  data-title="{{ "Theme modules"|trans({}, 'Admin.Modules.Feature') }}"
-                  data-content="{{ "Each theme you install will come with its own set of modules. You’ll find here all the modules related to your active theme."|trans({}, 'Admin.Modules.Help') }}">
-                </span>
+          {% if category.modules is empty %}
+            <div class="modules-list module-list-empty" data-name="{{ category.tab }}">
+              <p>
+                {{ 'You do not have module in « %categoryName% ».' | trans({'%categoryName%': category.name}, 'Admin.Modules.Feature') }}<br/>
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: What is the behavior we expect here?

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig [hunk 10]
```diff
@@ -1,107 +1,77 @@
...
+                {{ 'Discover the best-selling modules of this category in the %link% page.' | trans({'%link%': category.name}, 'Admin.Modules.Feature') }}
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: Same here, thanks!

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig [hunk 11]
```diff
@@ -1,107 +1,77 @@
...
     </div>
+  </div>
 {% endblock %}
 
 {% set js_translatable = {
-"Bulk Action - One module minimum": "You need to select at least one module to use the bulk action."|trans({}, 'Admin.Modules.Notification'),
-"Bulk Action - Request not found": 'The action "[1]" is not available, impossible to perform your request.'|trans({}, 'Admin.Modules.Notification'),
-"Bulk Action - Request not available for module": "The action [1] is not available for module [2]. Skipped."|trans({}, 'Admin.Modules.Notification'),
-}|merge(js_translatable) %}
+  "Bulk Action - One module minimum": "You need to select at least one module to use the bulk action."|trans({}, 'Admin.Modules.Notification'),
```
Comment (LouiseBonnard) [inline src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig:73]: Same here, sorryyy.

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig (hunk 3 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig]: Okay, I get it. Why is there an upper case at the beginning of the second word?

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 10 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: At this point, the user needs to click on the button 'Add new module', right?

File: src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php (hunk 1 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php:133]: It was a Lou request :)

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig (hunk 3 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig]: Same as Mock up :/ Don't understand why.

File: src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php (hunk 1 above)
Comment (LouToma) [reply src/PrestaShopBundle/Controller/Admin/Improve/ModuleController.php:133]: Yes, no S in English

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig (hunk 3 above)
Comment (LouToma) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/dropdown_categories.html.twig]: Mathias used a capital, can you change to Recently used please ?

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 10 above)
Comment (LouToma) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: Module Catalog. Add new module does not exist anymore

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 11 above)
Comment (LouToma) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig:73]: Behaviour remains the same as before

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 9 above)
Comment (LouToma) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: We want to inform merchants that they don't have any module in the category so they should go to the page "module catalog" to see more products they can add to their store

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 8 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: Categories came from Addons API.
`Administration` for example.

File: src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig (hunk 6 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Resources/views/Admin/Module/Includes/see_more.html.twig]: SIR YES SIR!

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 9 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: Well, in this case, I would write it like that: _You do not have any module in the %category_name% category._

File: src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig (hunk 10 above)
Comment (LouiseBonnard) [reply src/PrestaShopBundle/Resources/views/Admin/Module/manage.html.twig]: _Discover this category's best-selling modules in the %link% page._

File: install-dev/langs/en/data/tab.xml [hunk 12]
```diff
@@ -129,5 +129,5 @@
     <tab id="Warehouses" name="Warehouses" />
     <tab id="Webservice" name="Webservice" />
     <tab id="Zones" name="Zones" />
-    <tab id="Marketing_tools" name="Modules Selections" />
+    <tab id="Marketing_tools" name="Module Selections" />
```
Comment (Quetzacoalt91) [inline install-dev/langs/en/data/tab.xml:132]: I already asked about that string. The previous one was correct.

Comment (LouiseBonnard) [reply install-dev/langs/en/data/tab.xml:132]: cf. the famous "s" question about the feature's title. ;-)

File: src/PrestaShopBundle/Resources/views/Admin/Product/CatalogPage/Blocks/filters.html.twig [hunk 13]
```diff
@@ -76,7 +76,7 @@
             class="d-inline-block"
             bulkurl="{{ path('admin_product_bulk_action', {'action': 'activate_all'}) }}"
             massediturl="{{ path('admin_product_mass_edit_action', {'action': 'sort'}) }}"
-            redirecturl="{{ path('admin_product_catalog', {'limit': limit, 'offset': offset, 'orderBy': orderBy, 'sortOrder': sortOrder}) }}"
+$            redirecturl="{{ path('admin_product_catalog', {'limit': limit, 'offset': offset, 'orderBy': orderBy, 'sortOrder': sortOrder}) }}"
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Resources/views/Admin/Product/CatalogPage/Blocks/filters.html.twig]: Unexpected `$`

File: install-dev/langs/en/data/tab.xml (hunk 12 above)
Comment (PierreRambaud) [reply install-dev/langs/en/data/tab.xml:132]: Lou asked me to changed it ^^

Comment (Quetzacoalt91) [reply install-dev/langs/en/data/tab.xml:132]: Glad I asked :man_shrugging:

Comment (LouToma) [reply install-dev/langs/en/data/tab.xml:132]: No S no S :D cf Mock up

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 14]
```diff
@@ -0,0 +1,1099 @@
...
+
+const $ = window.$;
+
+/**
+ * Module Admin Page Controller.
+ * @constructor
+ */
+class AdminModuleController {
+  /**
+   * Initialize all listners and bind everything
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/module/controller.js]: `listeners`

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 15]
```diff
@@ -0,0 +1,1099 @@
...
+      'click',
+      self.categoryItemSelector,
+      function initializeCategorySelectClick() {
+        // Get data from li DOM input
+        self.currentRefCategory = $(this).data('category-ref');
+        self.currentRefCategory = self.currentRefCategory ? String(self.currentRefCategory).toLowerCase() : null;
+        // Change dropdown label to set it to the current category's displayname
+        $(self.categorySelectorLabelSelector).text($(this).data('category-display-name'));
+        $(self.categoryResetBtnSelector).show();
+        // Do Search on categoryRef
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/module/controller.js]: I dont understand this comment. Is it related to `self.updateModuleVisibility()` below ?

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 16]
```diff
@@ -0,0 +1,1099 @@
...
+      });
+    });
+
+    self.addonsCardGrid = $(this.addonItemGridSelector);
+    self.addonsCardList = $(this.addonItemListSelector);
+    self.updateModuleVisibility();
+    $('body').trigger('moduleCatalogLoaded');
+  }
+
+  updateModuleVisibility() {
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/module/controller.js]: This function is very long which makes it hard to understand

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 17]
```diff
@@ -0,0 +1,1099 @@
...
+          self.switchSortingDisplayTo(switchTo);
+          self.currentDisplay = switchTo;
+        }
+      }
+    );
+  }
+
+  switchSortingDisplayTo(switchTo) {
+    if (switchTo !== this.DISPLAY_GRID && switchTo !== this.DISPLAY_LIST) {
+      console.error(`Can't switch to undefined display property "${switchTo}"`);
```
Comment (matks) [inline admin-dev/themes/new-theme/js/pages/module/controller.js:1100]: 👍

File: admin-dev/themes/new-theme/js/pages/module/controller.js (hunk 15 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js]: Was the old comments from the previous code :/

File: admin-dev/themes/new-theme/js/pages/module/controller.js (hunk 16 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js]: Was the old function, just add little condition :/

File: admin-dev/themes/new-theme/js/pages/module/controller.js (hunk 15 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js]: btw I will change it

File: admin-dev/themes/new-theme/js/pages/module/controller.js (hunk 16 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js]: btw I will refactor it

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php [hunk 18]
```diff
@@ -26,15 +26,21 @@
 
 namespace PrestaShopBundle\Service\DataProvider\Admin;
 
+use PrestaShop\PrestaShop\Adapter\Module\Module as ApiModule;
 use GuzzleHttp\Exception\RequestException;
 use PrestaShopBundle\Service\DataProvider\Marketplace\ApiClient;
 use Psr\Log\LoggerInterface;
+use stdClass;
```
Comment (matks) [inline src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php:33]: I might be wrong but I do not see it being used in this page 😮 ?

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php [hunk 19]
```diff
@@ -72,42 +83,75 @@ public function getCategoriesMenu($modules)
...
-
-                foreach ($modules as $module) {
-                    $moduleCategory = $module->attributes->get('categoryName');
-                    $moduleCategoryParent = $this->getParentCategory($moduleCategory);
-
-                    if ($moduleCategoryParent === $categoryName) {
-                        $moduleIds[] = $module->attributes->get('id');
-                    }
+            $categoriesListing = $this->getCategories();
+            $categories = $this->initializeCategories($categoriesListing);
```
Comment (matks) [inline src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php:92]: 👍

**Comment by matks** [comment]
@PierreRambaud I did a review but I think a 2nd one is required as this topic is critical

File: src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php (hunk 18 above)
Comment (PierreRambaud) [reply src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php:33]: it is used on line 109.

`     * @param array|stdClass $categoriesListing Category listing`

File: admin-dev/themes/new-theme/js/pages/module/controller.js [hunk 20]
```diff
@@ -390,7 +431,7 @@ class AdminModuleController {
 
     const counter = {};
 
-    for (let i = 0; i < self.modulesList.length; i++) {
+    for (let i = 0; i < self.modulesList.length; i += 1) {
```
Comment (Quetzacoalt91) [inline admin-dev/themes/new-theme/js/pages/module/controller.js]: Could you store the value of `self.modulesList.length` outside? It is not cached and calculated at each loop.

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/pages/module/controller.js]: Done

**Review by Quetzacoalt91 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by mbadrani** [comment]
@PierreRambaud in issue #10048 it's we have "Note that "recently used does not appear, unless it has been selected in the "select" filter."
...but
With "recently used" selected in filter
image
Without "recently used" selected in filter
image

any idea?

**Comment by matks** [comment]
@mbadrani
`Note that 'recently used' does not appear, unless it has been selected in the "select" filter.`
=> it means the category should appear on the page only if selected
=> this is about what is displayed in the page content, not inside the drop-down menu

**Comment by PierreRambaud** [comment]
@mbadrani I don't understand what you mean about selected filter and the menu you show me above. How we can select a filter if it's hide? :D Can you have a check this @LouToma please? =)

**Comment by mbadrani** [comment]
@PierreRambaud look at my first screenshot ;-)

**Comment by LouToma** [comment]
If Pierre cannot solve the issue, it's fine.

**Comment by mbadrani** [comment]
Seen with @LouToma 
1) "recently used" appears twice if selected (look at the first screenshot)
2) "natif" won't be present in the list as it has been defined 
3) "theme module" is missing

**Comment by PierreRambaud** [comment]
@matks rebased!

**Comment by PierreRambaud** [comment]
1. It's the dropdown label like the old behavior, not appeared twice. If you select `Product` it will display `Product` for example, to show you what you selected.
2. It's an Addons API problem, @LouToma already knows what she needs to do :trollface: 
3. Theme module only appears when there is no connection between the BO and Addons API. Otherwise the categories you see come from Addons API.

**Comment by mbadrani** [comment]
@PierreRambaud 
"Theme Bundle" 
image

**Comment by PierreRambaud** [comment]
I amend my latest commit, this is why you still see the old name ;)

**Review by matks — DISMISSED** [review]

**Comment by mbadrani** [comment]
good job

**Comment by Quetzacoalt91** [comment]
Rebased for assets.

**Review by Quetzacoalt91 — APPROVED** [review]
Reapplying approval

**Comment by PierreRambaud** [comment]
Thanks me! :trollface:

**Pull request merged**